web/src/app/projects/view/view.component.ts

```diff
@@ -74,26 +71,8 @@ export class ViewProjectComponent implements OnInit, OnDestroy {
       .subscribe();
   }
 
-  // This function delete the project
   delete(): void {
-    let dialogConfig: MatDialogConfig = new MatDialogConfig();
-    dialogConfig = {
-      width: '500px',
-      data: {
-        title: 'Delete Project',
-        content: 'Are you sure you want to delete?',
-      },
-    };
-    this.dialogRef = this.dialog.open(DialogConfirmationComponent, dialogConfig);
-    this.dialogRef.afterClosed()
-      .subscribe((result: boolean) => {
-        if (result === true) {
-          this.projectSubscription.unsubscribe();
-          this.deleteSubscription = this.projectService
-            .delete(this.project.uid)
-            .subscribe(() => this.router.navigate(['/projects']));
-        }
-      });
+    this.projectService.showDeleteDialog(this.project.uid);
```

Comment:
On the `subscribe` here, the redirect make sense because the project will no longer exist